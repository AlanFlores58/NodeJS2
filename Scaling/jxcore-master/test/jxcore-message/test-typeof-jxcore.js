// Copyright & License details are available under JXCORE_LICENSE file

// gets members of an object and returns as sorted array
var getMembersNames = function(obj, name) {
  var arr = [];

  for(var o in obj) {
    if (obj.hasOwnProperty(o))
      arr.push(o);
  }

  arr.sort();

  // printing
  console.log("\nnamespace:", name);
  for(var o = 0, len = arr.length; o < len; o++) {
    var _name = arr[o];
    console.log("\ttypeof", _name, "=", typeof obj[_name]);
  }
};

getMembersNames(jxcore, "jxcore");
getMembersNames(jxcore.monitor, "jxcore.monitor");
getMembersNames(jxcore.store, "jxcore.store");
getMembersNames(jxcore.tasks, "jxcore.tasks");
getMembersNames(jxcore.utils, "jxcore.utils");
