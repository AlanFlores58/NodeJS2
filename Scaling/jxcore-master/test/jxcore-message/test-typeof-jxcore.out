
namespace: jxcore
	typeof embeddedModule = object
	typeof monitor = object
	typeof store = object
	typeof tasks = object
	typeof utils = object

namespace: jxcore.monitor
	typeof checkMonitorExists = function
	typeof checkPath = function
	typeof followMe = function
	typeof leaveMe = function
	typeof releasePath = function
	typeof respawned = number
	typeof restartMonitor = function
	typeof startIfNotExists = function
	typeof startMonitor = function
	typeof stopMonitor = function

namespace: jxcore.store
	typeof exists = function
	typeof get = function
	typeof read = function
	typeof remove = function
	typeof set = function
	typeof shared = object

namespace: jxcore.tasks
	typeof _addTask = function
	typeof _runOnThread = function
	typeof _runOnce = function
	typeof addTask = function
	typeof begin = function
	typeof emit = function
	typeof forceGC = function
	typeof getThreadCount = function
	typeof jobCount = function
	typeof killThread = function
	typeof on = function
	typeof register = function
	typeof runOnThread = function
	typeof runOnce = function
	typeof setThreadCount = function
	typeof unloadThreads = function

namespace: jxcore.utils
	typeof OSInfo = function
	typeof argv = object
	typeof bufferID = function
	typeof cmdSync = function
	typeof console = object
	typeof continue = function
	typeof getCPU = function
	typeof getOS = function
	typeof isFunction = function
	typeof jump = function
	typeof pause = function
	typeof smartRequire = function
	typeof uniqueId = function
